### PR TITLE
Improve Onboarding

### PR DESCRIPTION
**Pull request by shervElmi**
### PR Description

This PR enhances and updates the existing Formidable Forms Onboarding Wizard for improved functionality and user experience.

### QA URL
https://qa.formidableforms.com/sherv/wp-admin/admin.php?page=formidable-onboarding-wizard

### Demo Video
https://share.cleanshot.com/2L0v82hf

https://github.com/user-attachments/assets/6a3d1ead-f4b1-4950-9d57-91350368d1fa

**_To easily load the onboarding wizard after having already activated Formidable before is to load `/wp-admin/admin.php?page=formidable-onboarding-wizard` directly_**

### CODE REVIEW

File: js/src/onboarding-wizard/ui/rootline.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * Updates the rootline to reflect the current and completed steps.
+ *
+ * - Applies COMPLETED_STEP_CLASS to steps before the current one.
+ * - Applies CURRENT_CLASS to the current step, unless it is the success step.
+ *
+ * @param {string} currentStep The current step in the process.
+ * @return {void}
+ */
+export function updateRootline( currentStep ) {
+	if (currentStep === STEPS.UNSUCCESSFUL) {
```
Comment (Crabcyborg) [inline js/src/onboarding-wizard/ui/rootline.js:24]: We haven't gotten linting set up yet, but I'd generally try to have spaces inside of these braces.

File: classes/views/onboarding-wizard/footer.php [hunk 2]
```diff
@@ -11,15 +11,22 @@
...
 	<a <?php FrmAppHelper::array_to_html_params( $secondary_button_attributes, true ); ?>>
 		<?php echo esc_html( $args['secondary-button-text'] ); ?>
 	</a>
+
 	<a <?php FrmAppHelper::array_to_html_params( $primary_button_attributes, true ); ?>>
-		<?php echo esc_html( $args['primary-button-text'] ); ?>
+		<?php
+		echo esc_html( $args['primary-button-text'] );
+
+		if ( $args['primary-button-with-icon'] === true ) {
```
Comment (Crabcyborg) [inline classes/views/onboarding-wizard/footer.php:27]: Do we expect other values here other than `true` and `false`? If `$args['primary-button-with-icon']` is consistently a boolean, I'd generally opt to not include the `=== true` which seems redundant.

**Review by Crabcyborg — COMMENTED** [review]
I haven't looked much into the actual changes to onboarding yet, but the code looks good.

I left some pretty minor comments.

I'll test this out soon. Thanks Sherv!

File: classes/controllers/FrmOnboardingWizardController.php [hunk 3]
```diff
@@ -279,41 +279,31 @@ public static function render() {
...
 	/**
-	 * Handle AJAX request to setup the "Default Email Address" step.
+	 * Handle AJAX request to setup the "Never miss an important update" step.
 	 *
 	 * @since 6.9
 	 *
 	 * @return void
 	 */
-	public static function ajax_setup_email_step() {
+	public static function ajax_consent_tracking() {
```
Comment (Crabcyborg) [inline classes/controllers/FrmOnboardingWizardController.php:298]: @shervElmi As far as I understand, we want to also subscribe to ActiveCampaign at this step?

The "Get key updates, tips, and occasional offers to enhance your WordPress experience." part of this onboarding isn't related to the consent.

We'll need to figure this out still.

Comment (Crabcyborg) [reply classes/controllers/FrmOnboardingWizardController.php:298]: I just made a small commit that adds the active campaign subscribe https://github.com/Strategy11/formidable-forms/pull/2042/commits/f3eebb74d69008961509a8159dabc1adfcbb31c3

Comment (shervElmi) [reply classes/controllers/FrmOnboardingWizardController.php:298]: That sounds great! Thanks @Crabcyborg 🚀

**Review by Crabcyborg — APPROVED** [review]
I think this looks good.

I also requested Laura's review so this can get some extra testing.

Thanks Sherv!

**Comment by shervElmi** [comment]
Hi @lauramekaj1,

Thanks for the feedback, I addressed that issue. Could you confirm my solution?

**Comment by lauramekaj1** [comment]
> Hi @lauramekaj1,
> 
> Thanks for the feedback, I addressed that issue. Could you confirm my solution?

Hi @shervElmi, I checked and seems that it's fixed now. Thank you!

**Review by lauramekaj1 — APPROVED** [review]

**Comment by Crabcyborg** [comment]
Thanks Laura!

🚀

**Pull request merged**